kubevirtci/sync.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source ./hack/common.sh
+
+# Please note that the validation webhooks are not currently deployed so take care when using a custom SSP CR
+export KUBEVIRT_SSP=${KUBEVIRT_SSP:-../config/samples/ssp_v1beta1_ssp.yaml}
+
+# The ssps.ssp.kubevirt.io CRD might not be present if this is the first sync so check before attempting to delete any actual objects
+if ./kubevirtci/kubevirtci.sh kubectl get crd/ssps.ssp.kubevirt.io; then
+    ./kubevirtci/kubevirtci.sh kubectl delete --ignore-not-found=true ssps -n kubevirt --all
+fi
+# The remaining CRDs should be present already from the k8s and KubeVirt installs so just ignore object not found failures
+./kubevirtci/kubevirtci.sh kubectl delete --ignore-not-found=true deployment/ssp-operator -n kubevirt
+./kubevirtci/kubevirtci.sh kubectl delete --ignore-not-found=true virtualmachineclusterinstancetype --all
+./kubevirtci/kubevirtci.sh kubectl delete --ignore-not-found=true virtualmachineclusterpreference --all
+
+nodes=()
+for i in $(seq 1 "${KUBEVIRT_NUM_NODES}"); do
+    nodes+=("node$(printf "%02d" "${i}")")
+done
+
+# TODO - clean this up
+localhost_img_registery=$(./kubevirtci/kubevirtci.sh registry)
+IMG_REPOSITORY=$localhost_img_registery IMG=$localhost_img_registery:latest make container-build container-push
+
+for node in ${nodes[@]}; do
+    ./kubevirtci/kubevirtci.sh ssh "${node}" "echo registry:5000/kubevirt/ssp-operator | xargs \-\-max-args=1 sudo docker pull"
```

Comment:
I saw other versions of this snippet using podman instead (or checking at runtime which one to use). This might depend on the used k8s provider?

Reply:
The only code I've seen where podman was used was downstream OCP/OKD specific, I'll check again.

Reply:
That might be the case, just wanted to give you a heads up. I thought someone was working on replacing docker in kubevirtci nodes as well.

Reply:
Turns out we actually use podman inside the node VMs but it is also symlinked to `/usr/bin/docker`.

https://github.com/kubevirt/kubevirtci/blob/c1579907e89a5fbec9f7eab45d6287b11094ed3c/cluster-provision/k8s/1.25/provision.sh#L133